syntax = "proto3";

// DataListResponse is the response message for the Authentication method from the chat service.
//
// It contains a flag to indicate if the message was successfully sent and an error message if there was an error.
// It also sends a list of strings as the data to return
message DataListResponse {
    // success is a flag to indicate if the message was successfully sent.
    bool success = 1;
    // http status is the integer response of the request results whilst following the http code format
    int32 http_status = 2;
    // error is the error message if there was an error sending the message, it is empty if the message was successfully sent.
    repeated string error_message = 3;
    // user_id is the id to return when auth is complete
    repeated string data = 4;
}

message DegreeRequest {
}

message TagRequest {
}


service DataFetch {
    // FetchDegrees returns all degrees found
    rpc FetchDegrees(DegreeRequest) returns (DataListResponse) {}
    // FetchTags returns all tags found
    rpc FetchTags(TagRequest) returns (DataListResponse) {}
}
